torch/distributions/distribution.py

```diff
@@ -10,10 +10,15 @@ class Distribution(object):
 
     has_rsample = False
     has_enumerate_support = False
+    support = None
+    params = {}
 
     def __init__(self, batch_shape=torch.Size(), event_shape=torch.Size()):
         self._batch_shape = batch_shape
         self._event_shape = event_shape
+        for param, constraint in self.params.items():
+            if not constraint.check(self.__getattribute__(param)):
```

Comment:
This shold be `constraint.check(...).all()`, since constraint checks are elementwise.